backend/src/apiserver/main.go

```diff
@@ -189,6 +195,19 @@ func loadSamples(resourceManager *resource.ResourceManager) error {
 		if configErr != nil {
 			return fmt.Errorf("Failed to decompress the file %s. Error: %v", config.Name, configErr)
 		}
+		// Patch the default bucket name read from ConfigMap
+		if common.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
+			defaultBucket := common.GetStringConfig(DefaultBucketNameEnvVar)
+			projectId := common.GetStringConfig(ProjectIDEnvVar)
+			patchMap := map[string]string{
+				"<your-gcs-bucket>": defaultBucket,
+				"<your-project-id>": projectId,
+			}
+			pipelineFile, err = server.PatchPipelineDefaultParameter(pipelineFile, patchMap)
+			if err != nil {
+				return fmt.Errorf("Failed to patch default value to %s. Error: %v", config.Name, err)
```

Comment:
If patching fails, will APIserver keep restarting?

Reply:
Yes it does. Do you recommend swallow the error? IMO patching should work unless user configured something in the wrong way. Thanks!